test/mpi/spawn/join.c

```diff
@@ -88,6 +88,8 @@ int main(int argc, char *argv[])
         MPI_Send(hostname, namelen + 1, MPI_CHAR, 0, 0, MPI_COMM_WORLD);
         MPI_Send(&port, 1, MPI_INT, 0, 1, MPI_COMM_WORLD);
 
+        MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
Hmm, yes you are right. The `accept` shouldn't start until the data has arrived on the other side. Though again, the commit message is misleading a bit here. The problem is the test is not conservative enough for a compliant implementation of blocking `MPI_Send`. It is not specific to CH4/OFI, or even OFI. Let's try to clear that up.